### ISSUE TITLE
Improve build mechanism

### ISSUE THREAD

**Issue by pavanky** [arrayfire/arrayfire-rust#7]
Would it be possible to pass in command line arguments to cargo instead of editing build.rs to configure the following:

- Changing the backend
- Changing the number of threads for make

It should also be possible to use arrayfire that is already installed on the system.

**Comment by jramapuram** [comment]
Yup, first point should be easy. Just wanted to get something up right away.
Second can be done as well, however it is OS dependent. I.e finding packages on OSX vs. Linux, etc. and thus a little trickier.

**Comment by pavanky** [comment]
@jramapuram If we can pass command line arguments to `build.rs`, I can make the build process use the system libraries.

**Comment by jramapuram** [comment]
Hmm, you want to pass in paths to the libs as opposed to auto-discover?

The Rust compiler actually is pretty awesome. 
The two lines that need modifying are:
```rust
    println!("cargo:rustc-link-search=native={}", build_dir.join("src/backend/cuda").display());
    println!("cargo:rustc-link-lib=dylib=afcuda");
```

**Comment by jramapuram** [comment]
https://github.com/arrayfire/arrayfire_rust/issues/7 solves using existing libs

**Comment by 9prady9** [comment]
After #29 , this change is not needed and use of existing libs is taken care of by another PR. Hence, closing this issue.

**Comment by jramapuram** [comment]
:+1:
